### DISCUSSION TITLE
The part of 'Extract ECG Derived Respiration (EDR)

### DISCUSSION THREAD

**Discussion by linestao** [neuropsychology/NeuroKit#598] [Q&A]
I encountered some difficulties in extracting EDR with ECG。I want to know if the part of 'Extract ECG Derived Respiration (EDR)'  will continue to be updated。
Thank you for your code's help to my work, and thank you for your reply in your busy schedule

**Comment by zen-juen** [comment]
Hi @linestao, for us to investigate further, do you mean you encountered problems when running this [example docs](https://neurokit2.readthedocs.io/en/latest/examples/edr.html) or with your own script? If the latter, could you share with us a sample of your data and a reproducible script?

**Reply by linestao** [reply]
There are such comments in your code:
_def ecg_rsp(ecg_rate, sampling_rate=1000, method="vangent2019"):
    """Extract ECG Derived Respiration (EDR).
    This implementation is far from being complete, as the information in the related papers
    prevents me from getting a full understanding of the procedure. Help is required!_

So I guess your team has temporarily shelved the work of ECG signal extraction EDR

**Reply by DominiqueMakowski** [reply]
https://github.com/neuropsychology/NeuroKit/blob/d01111b9b82364d28da01c002e6cbfc45d9493d9/neurokit2/ecg/ecg_rsp.py#L7-L8

I think we can remove these lines (I feel like it applies (applied?) to one of the method) and replace by a smol description of what EDR is.

**Comment by paudel54** [comment]
why does the rate goes a negative value? Does beating rate being negative even exist?
